src/linux/init/WSLAInit.cpp

```diff
@@ -159,12 +159,16 @@ void HandleMessageImpl(wsl::shared::SocketChannel& Channel, const WSLA_TTY_RELAY
 {
     THROW_LAST_ERROR_IF(fcntl(Message.TtyMaster, F_SETFL, O_NONBLOCK) < 0);
```

Comment:
>TtyMaster [](http://example.com/codeflow?start=38&length=9)

What controls the lifetime of these file descriptors?

Reply:
Right now they're tied to the process that's allocating them. When the channel is closed from the service side, the process exits and all its fd's are closed